### PR TITLE
[RBAC] Handle authZ for filters with references

### PR DESCRIPTION
**Pull request by dirkkul**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/grpc/v1/filters.go [hunk 1]
```diff
@@ -194,9 +194,9 @@ func extractDataTypeProperty(getClass func(string) *models.Class, operator filte
 		}
 
 		classOfProp := on[len(on)-2]
-		class := getClass(classOfProp)
-		if class == nil {
-			return dataType, fmt.Errorf("could not find class %s in schema", classOfProp)
+		class, err := getClass(classOfProp)
+		if err != nil {
+			return schema.DataTypeString, err
```
Comment (tsmith023) [inline adapters/handlers/grpc/v1/filters.go:199]: Should this not be `return dataType, err` to align with how the other error branches are defined in this func?

File: adapters/handlers/grpc/v1/filters.go [hunk 2]
```diff
@@ -210,7 +210,10 @@ func extractDataTypeProperty(getClass func(string) *models.Class, operator filte
 			return schema.DataTypeInt, nil
 		}
 
-		class := getClass(className)
+		class, err := getClass(className)
+		if err != nil {
+			return schema.DataTypeString, err
```
Comment (tsmith023) [inline adapters/handlers/grpc/v1/filters.go:215]: Same as above

File: adapters/handlers/grpc/v1/filters.go [hunk 3]
```diff
@@ -257,10 +260,10 @@ func extractPath(className string, on []string) (*filters.Path, error) {
...
 }
 
-func extractPathNew(getClass func(string) *models.Class, className string, target *pb.FilterTarget, operator filters.Operator) (*filters.Path, schema.DataType, error) {
-	class := getClass(className)
-	if class == nil {
-		return nil, "", fmt.Errorf("could not find class %s in schema", className)
+func extractPathNew(getClass func(string) (*models.Class, error), className string, target *pb.FilterTarget, operator filters.Operator) (*filters.Path, schema.DataType, error) {
+	class, err := getClass(className)
+	if err != nil {
+		return nil, schema.DataTypeString, err
```
Comment (tsmith023) [inline adapters/handlers/grpc/v1/filters.go:266]: Same here, why the change from `return nil, "", err` to `return nil, schema.DataTypeString, err`?

File: adapters/handlers/grpc/v1/parse_search_request.go [hunk 4]
```diff
@@ -330,12 +330,16 @@ func (p *Parser) Search(req *pb.SearchRequest, config *config.Config) (dto.GetPa
 	}
 
 	if req.Filters != nil {
-		clause, err := ExtractFilters(req.Filters, p.getClass, req.Collection)
+		// Todo: Replace with real auth function
```
Comment (tsmith023) [inline adapters/handlers/grpc/v1/parse_search_request.go:333]: Will this TODO be handled in another PR?

**Review by tsmith023 — APPROVED** [review]

File: adapters/handlers/grpc/v1/parse_search_request.go (hunk 4 above)
Comment (dirkkul) [reply adapters/handlers/grpc/v1/parse_search_request.go:333]: yes, this requires more changes at other places and I wanted to minimize the amount of cahnges in one PR :)
